src/librustc/session/config.rs

```diff
@@ -926,6 +927,18 @@ macro_rules! options {
             }
         }
 
+        fn parse_opt_comma_list(slot: &mut Option<Vec<String>>, v: Option<&str>)
+                      -> bool {
+            match v {
+                Some(s) => {
+                    let v = s.split(',').map(|s| s.to_string()).collect();
+                    *slot = Some(v);
```

Comment:
What will this do if we see the flag multiple times?

I think we have this sort of problem all over the place, I would prefer if `-Z allow_features=foo -Z allow_features=bar` worked like `-Z allow_features=foo,bar`.

cc @rust-lang/compiler Should we do an audit? It's plausible fixing some of the other flags to do "the right thing" when provided multiple times may be a breaking change.

At least this one is unstable though.

Reply:
If the flag is supplied multiple times the last invocation will override the previous ones. This is (afaict) the behavior of all the -Z flags that take a list of arguments.

Reply:
Yes and I'm saying it's not intentional most of the time, it just arises from how the CLI arguments are parsed.

I wonder what would happen were we to try to switch to e.g. `structopt`.

Reply:
Sorry, was responding in answer to your original "What will this do if we see the flag multiple times?"-- I agree that trying to shift everything over to a different behavior would be a worthwhile thing to investigate.

Reply:
Opened #59219